models.py

```diff
@@ -6,9 +6,49 @@
 
 from data_pipeline.glove_embeddings import get_weight_matrix
 
+# this size came from the highest code of a character that was found in the data set
+# の has the code 12398
+NUMBER_OF_ASCII_CHARACTERS = 12400
+
 #TODO: move this.
 BOS_IDX = 1
 
+class CharacterLevelEmbedding(nn.Module):
+
+
+  def __init__(self, config: CfgNode):
+    super(CharacterLevelEmbedding, self).__init__()
+    self.embedding = nn.Embedding(NUMBER_OF_ASCII_CHARACTERS, config.CHAR_EMB_DIM)
+    self.gru = nn.GRU(config.CHAR_EMB_DIM, config.CHAR_HIDDEN_SIZE)
+    self.char_hidden_size = config.CHAR_HIDDEN_SIZE
+
+  def forward(self, inputs: torch.Tensor, input_lengths: torch.Tensor):
+    """
+    Compute the learned embeddings at character level
+       Args:
+           inputs: (torch.Tensor): Inputs (word len, input seq len, batch size).
+           input_lengths: (torch.Tensor): Inputs Length (input seq len, batch size)
+       Returns:
+           output: the embedded input sequence
+    """
+    # word_len x input_seq_len x batch_size x char_emb_dim
+    embedded_inputs = self.embedding(inputs)
+    word_len, input_seq_len, batch_size, char_emb_size = embedded_inputs.shape
+    # view will return a tensor with the same data but with a different shape
+    # the reshape and view are used in order to feed the embedded_inputs and inputs_length to the pack_padded_sequence
```

Comment:
the comments are not explaining much in the context of your code...
I would expect smth like "Use view to go from X shape to Y shape" -- that would be more helpful than just putting something one can find in the documentation but not really explain how you use it and why